src/S3File.cc

```diff
@@ -48,81 +48,26 @@ XrdVERSIONINFO(XrdOssGetFileSystem, S3);
 S3File::S3File(XrdSysError &log, S3FileSystem *oss)
 	: m_log(log), m_oss(oss), content_length(0), last_modified(0) {}
 
-int parse_path(const S3FileSystem &fs, const char *fullPath,
-			   std::string &exposedPath, std::string &object) {
-	//
-	// Check the path for validity.
-	//
-	std::filesystem::path p(fullPath);
-	auto pathComponents = p.begin();
-
-	// Iterate through components of the fullPath until we either find a match
-	// or we've reached the end of the path.
-	std::filesystem::path currentPath = *pathComponents;
-	while (pathComponents != p.end()) {
-		if (fs.exposedPathExists(currentPath.string())) {
-			exposedPath = currentPath.string();
-			break;
-		}
-		++pathComponents;
-		if (pathComponents != p.end()) {
-			currentPath /= *pathComponents;
-		} else {
-			return -ENOENT;
-		}
-	}
-
-	// Objects names may contain path separators.
-	++pathComponents;
-	if (pathComponents == p.end()) {
-		return -ENOENT;
-	}
-
-	std::filesystem::path objectPath = *pathComponents++;
-	for (; pathComponents != p.end(); ++pathComponents) {
-		objectPath /= (*pathComponents);
+int S3File::Open(const char *path, int Oflag, mode_t Mode, XrdOucEnv &env) {
+	if (m_log.getMsgMask() & XrdHTTPServer::Debug) {
+		m_log.Log(LogMask::Warning, "S3File::Open", "Opening file", path);
 	}
-	object = objectPath.string();
-
-	fprintf(stderr, "object = %s\n", object.c_str());
-
-	return 0;
-}
 
-int S3File::Open(const char *path, int Oflag, mode_t Mode, XrdOucEnv &env) {
 	std::string exposedPath, object;
-	int rv = parse_path(*m_oss, path, exposedPath, object);
+	auto rv = m_oss->parsePath(path, exposedPath, object);
 	if (rv != 0) {
 		return rv;
 	}
-	if (!m_oss->exposedPathExists(exposedPath))
+	auto ai = m_oss->getS3AccessInfo(exposedPath);
+	if (!ai) {
 		return -ENOENT;
-
-	std::string configured_s3_region = m_oss->getS3Region(exposedPath);
-	std::string configured_s3_service_url = m_oss->getS3ServiceURL(exposedPath);
-	std::string configured_s3_access_key =
-		m_oss->getS3AccessKeyFile(exposedPath);
-	std::string configured_s3_secret_key =
-		m_oss->getS3SecretKeyFile(exposedPath);
-	std::string configured_s3_bucket_name = m_oss->getS3BucketName(exposedPath);
-	std::string configured_s3_url_style = m_oss->getS3URLStyle();
-
-	// We used to query S3 here to see if the object existed, but of course
-	// if you're creating a file on upload, you don't care.
-
-	this->s3_object_name = object;
```

Comment:
This section of bookeeping was removed, but the caused a bug because the s3_object_name still exists in the class, but never gets set. This causes a segfault and file handling errors.